### DISCUSSION TITLE
Scratch 2

### DISCUSSION THREAD

**Discussion by jdivins** [airsdk/Adobe-Runtime-Support#1805] [Q&A]
Hi,
Thank you for your work!
I need to run Scratch 2 on a mac book with Catalina 10.15.7 installed on it.
I need to use the Scratch 2 version because I have a plugin for leap motion that only runs on this version.
I have installed the last Adobe Air Runtime version from this website: https://airsdk.harman.com/runtime
And it is installed correctly but after installing Scratch 2 and opening it I get a white page:
![image](https://user-images.githubusercontent.com/1632358/160488573-eae86b8f-4524-4e71-9256-8ba7e2820df8.png)

I have running Scratch 2 on another mac os x computer with Adobe Air 32.0 and with Mac Os Catalina 10.15.7, maybe can I copy the files from one computer to the other one? But ... which files?

Or there are some other kinds of solutions?

Thank you very much for your help

**Comment by ajwfrost** [comment]
Odd that it installs and runs but is just blank ... 

I've just tried this on a Big Sur machine and it worked okay, the page comes up white for about a second before it gets populated by the normal Scratch UI. Are you able to try things like re-sizing the window in case that kicks it into life?

The other things I can think of that may be affecting it:

1) it's perhaps trying to download some initial elements or configuration from the internet, so I'm wondering if there's a failure in this. I assume the machine is online (and given it's Catalina, it must also be running on Intel hardware?) but am wondering if there are any app security updates preventing this. Can you check in Settings -> Security & Privacy to see if there are any notifications about things being blocked, or try turning off the firewall briefly whilst running it?

2) the other difference that I can see is you've got a localised version, although Scratch does appear to have support for a huge number of languages. But it may be worth a try to switch the computer language back to US-English and re-try?

Scratch uses the "direct" rendering method which means composition of the application happens via OpenGL on a mac, so worth checking that everything is working well with that - there's an app called "GLview" available from the app store that can show a summary of the OpenGL information..

But other than things like re-booting, re-installing, etc, that's about all I can think of! It should work but there are some quirks. Copying files from another computer may be possible but you may then end up hitting issues with security/code-signing requirements on Catalina...

thanks

**Reply by jdivins** [reply]
No luck.
I have tried GLView and I see the cube passing the various tests.
I have tried resizing and making the application full screen but it has remained blank.
I have tried changing the language and reinstalling, all the same white screen.
I have tried uninstalling AdobeAir Runtime and Scratch 2 with the same result.
Also I have checked if there is some blocked app on privacity and disabled firewall during installation and app loading. And there were blocks but after unblock theme I can open the app.
Can you share me the links you used to download scratch 2 and adobe air?

Thank you very much for your suggestions.

Any other advice?

**Reply by ajwfrost** [reply]
That's very weird. We do sometimes see problems with applications not opening, due to code signatures and security blocks, but to have it open up and remain a white screen is very odd.

The links I used were just:
https://scratch.mit.edu/download/scratch2
(the macOS link plus the Adobe AIR link which goes to https://airsdk.harman.com/runtime)

I am seeing the final one of their "known issues" so I'm wondering whether this is part of the problem:
> On Mac OS you may see a prompt indicating that "Scratch 2 is trying to install a new helper tool" and asking for your user name and password. We are currently investigating a solution to this problem.

But even if I hit 'cancel' on that, it's loading up correctly.

The next step to debug this would be to install Adobe Scout and run that, before launching Scout. Would you be able to try that, capture the log file and post it here?
https://helpx.adobe.com/uk/scout/kb/download-adobe-scout-cc.html

thanks

**Reply by jdivins** [reply]
You used same files than me.

Here you are:
[Scratch 2 Offline Editor.flm.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/8374293/Scratch.2.Offline.Editor.flm.zip)

As you will see I have maintained screen opened during a minute with blank screen and making some screen moves and resizing.

Thank you!

**Reply by ajwfrost** [reply]
Thanks .. so I can see things like an ioError happening and uncaughtError which may be related; it looks like it's trying to log these out somewhere but I've just looked on my own computer and can't see where the logs may be stored. If you look in the below folder though, do you see loads of folders/files?
`~/Library/Application Support/edu.media.mit.Scratch2Editor/Local Store`

The other thing which has been shown in the trace console and is more likely to be the problem is:
```
PermissionError: 'Error #3800'
```
This can be thrown when the app is trying to access things like camera/microphone, or even just a part of the filesystem.

Are you able to go into Settings -> Security & Privacy -> Privacy tab, select "Full disk access" and click on the lock to make changes, then the `+` button to add a new app. Browse for Applications -> Scratch 2 and you should see this appear in the list of enabled apps, with a ticked checkbox next to it. Click on the lock again to seal it back up and then re-try..?

thanks

**Reply by jdivins** [reply]
This is the content of the folder:
[Archive.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/8378246/Archive.zip)

I have enabled full disk acces for scratch 2:
<img width="437" alt="image" src="https://user-images.githubusercontent.com/1632358/160779691-c377ca8a-c5d3-41a4-8930-57f1b9371402.png">

But still blank screen.

These are new scout logs after your suggestion:
[Scratch 2 Offline Editor_after_fda.flm.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/8378267/Scratch.2.Offline.Editor_after_fda.flm.zip)


Again, THANK YOU very much for your help!

**Reply by ajwfrost** [reply]
Hmm... I have a lot more files in my library area, plus more information stored into the shared object. Plus, the permission error is still happening judging by that Scout log.

I don't suppose it is possible to open it up with root permissions? so e.g. if you go into a terminal, type:
```
sudo /Applications/Scratch\ 2.app/Contents/MacOS/Scratch\ 2
```
and enter your password when it asks you for it?

thanks

**Reply by jdivins** [reply]
Blank screen again:
sudo logs with scout in sudo:
[Scratch 2 Offline Editor_sudo.flm.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/8379506/Scratch.2.Offline.Editor_sudo.flm.zip)

**Reply by ajwfrost** [reply]
:-( still the permission error. Frustrating that we can't see more details about what is behind that ... we may need to create a custom AIR package in order to get to the bottom of this one!

**Reply by jdivins** [reply]
Can I help in some way to test it?

**Reply by ajwfrost** [reply]
Hi @jdivins  - we've created a custom build that you can download from the below link, are you able to try installing this and then use Scout to get another log of what's happening?
https://transfer.harman.com/message/tsfquMfHqmx3NUXiY7u0O2

thanks

**Reply by jdivins** [reply]
Hi @ajwfrost . I'm sorry but I hadn't seen any kind of notice and now the link seems to be broken.
I have reinstalled my operating system and I came to the same error. Can you resend me the build?
Thank you!

**Reply by ajwfrost** [reply]
@jdivins let me see if we've got that build still, I think the system deletes messages after a week or two.. if not we can look at re-creating it.
